src/main/java/net/openhft/chronicle/bytes/internal/Decimalizer.java

```diff
@@ -0,0 +1,217 @@
+        public boolean toDecimal(float value, DecimalAppender decimalAppender) {
+            return LITE.toDecimal(value, decimalAppender)
+                    || USES_BIG_DECIMAL.toDecimal(value, decimalAppender);
+        }
+    }
+
+    /**
+     * Light-weight implementation of Decimalizer, which uses a simple rounding approach for conversion.
+     */
+    class Lite implements Decimalizer {
+
+        public static final int LARGEST_EXPONENT_IN_LONG = 18;
+
+        /**
+         * Converts a double value to its decimal representation using a simple rounding approach and appends it.
+         *
+         * @param value           the double value to be converted.
+         * @param decimalAppender the DecimalAppender used to store the converted decimal value.
+         * @return true if the conversion and appending was successful, false otherwise.
+         */
+        public boolean toDecimal(double value, DecimalAppender decimalAppender) {
+            boolean isNegative = Double.doubleToLongBits(value) < 0;
+            double absValue = Math.abs(value);
+
+            long factor = 1;
+            for (int exponent = 0; exponent <= LARGEST_EXPONENT_IN_LONG; exponent++) {
+                long mantissa = Math.round(absValue * factor);
+                if ((double) mantissa / factor == absValue) {
+                    decimalAppender.append(isNegative, mantissa, exponent);
+                    return true;
+                }
+                factor *= 10;
+            }
+            return false;
+        }
+
+        /**
+         * Converts a float value to its decimal representation using a simple rounding approach and appends it.
+         *
+         * @param value           the float value to be converted.
+         * @param decimalAppender the DecimalAppender used to store the converted decimal value.
+         * @return true if the conversion and appending was successful, false otherwise.
+         */
+        public boolean toDecimal(float value, DecimalAppender decimalAppender) {
+            boolean sign = Double.doubleToLongBits(value) < 0;
+            float absValue = Math.abs(value);
+
+            long factor = 1;
+            for (int exponent = 0; exponent <= LARGEST_EXPONENT_IN_LONG; exponent++) {
+                long mantissa = Math.round(absValue * (double) factor);
+                if ((float) mantissa / factor == absValue) {
+                    decimalAppender.append(sign, mantissa, exponent);
+                    return true;
+                }
+                factor *= 10;
+            }
+            return false;
+        }
+    }
+
+    /**
+     * BigDecimal-based implementation of Decimalizer for higher precision conversions.
+     */
+    class UsesBigDecimal implements Decimalizer {
+
+        /**
+         * Field reference to the internal storage of BigDecimal values.
+         */
+        static final java.lang.reflect.Field INT_COMPACT = Jvm.getFieldOrNull(BigDecimal.class, "intCompact");
```

Comment:
Not worth using method handles for this access?

Reply:
I could explore if that makes a difference. Using Unsafe is probably the fastest, however, given this is using BigDecimal at all isn't the most performant.